### ISSUE TITLE
Analyzer doesn't consider macro expansion before flagging errors.

### ISSUE THREAD

**Issue by metasim** [intellij-rust/intellij-rust#8640]
## Environment

* **IntelliJ Rust plugin version:** 0.4.164.4409-213
* **Rust toolchain version:** 1.58.1 (db9d1b20b 2022-01-20) aarch64-apple-darwin
* **IDE name and version:** CLion 2021.3.3 (CL-213.6777.58)
* **Operating system:** macOS 12.2.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.macros.proc

## Problem description

I'm using the `extend` crate which eliminates boilerplate when writing extension methods. It auto-generates a `trait` from an `impl`.

This code

```rust
use extend::ext;

#[ext]
impl<T> Vec<T> {
    fn scramble(mut self) -> Self {
        let len = self.len();
        self.swap(0, len/2);
        self
    }
}

fn main() {
    let nums = vec![1, 2, 3, 4, 5];
    println!("{:?}", nums.scramble());
}
```

is flagged as an error

<img width="736" alt="image" src="https://user-images.githubusercontent.com/131013/157054752-e59499aa-fabe-472f-b43b-299100c4759f.png">

even though it compiles fine and runs:

```
cargo run --package ext-err --bin ext-err
    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
     Running `target/debug/ext-err`
[3, 2, 1, 4, 5]
```

## Steps to reproduce

`main.rs` and `Cargo.toml` available here:

https://gist.github.com/metasim/d05f7b305752a08411efc3674c61cc2f

**Comment by Undin** [comment]
Just enable `org.rust.cargo.evaluate.build.scripts` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) in addition to current ones.
See #6908 for more details
